cfme/tests/infrastructure/test_vm_power_control.py

```diff
@@ -583,3 +583,76 @@ def test_guest_os_shutdown(appliance, testing_vm_tools, ensure_vm_running, soft_
     new_last_boot_time = view.entities.summary("Power Management").get_text_of("Last Boot Time")
     soft_assert(new_last_boot_time == last_boot_time,
                 "ui: {} should ==  orig: {}".format(new_last_boot_time, last_boot_time))
+
+
+@pytest.fixture(params=['archived_vm', 'orphaned_vm'])
+def archive_orphan_vm(request, provider, testing_vm):
+    """This fixture is used to create archived or orphaned VM"""
+    if request.param == "archived_vm":
+        # Archive VM by retiring it
+        testing_vm.mgmt.delete()
+        testing_vm.wait_for_vm_state_change(desired_state='archived', timeout=720,
+                                            from_details=False, from_any_provider=True)
+    else:
+        # Orphan VM by removing provider from CFME
+        provider.delete_if_exists(cancel=False)
+        testing_vm.wait_for_vm_state_change(desired_state='orphaned', timeout=720,
+                                            from_details=False, from_any_provider=True)
+    yield request.param
```

Comment:
Consider changing this up a bit, to yield a tuple of the testing_vm and the request param. Also consider shortening the param to just `archived` or `orphaned`, dropping `_vm`